### ISSUE TITLE
PaletteEdit: Cannot edit or rename color by keyboard

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#58188]
The ColorPicker popover has a focus trap, and does not close on <kbd>Esc</kbd> in certain cases, making it impossible to exit out of the popover by keyboard.

## Reproduction steps (by keyboard)

1. Click the "Add color" button.
2. Focus is moved to the ColorPicker popover, and is stuck.

https://github.com/WordPress/gutenberg/assets/555336/a667d62c-f5d5-4b2e-9c1e-03f36714c7e7

Same thing happens when you go to the "Color options dropdown menu ▸ Show details" and choose a color to edit. You are stuck in a popover and cannot edit color values or names.

## Proposed fix

The ColorPicker popover should close on <kbd>Esc</kbd>.

**Comment by mirka** [comment]
Fixed by #62753
